drivers/serial/uart_nrfx_uart.c

```diff
@@ -640,8 +641,9 @@ static void tx_isr(const struct device *dev)
 	    !uart0_cb.tx_abort) {
 #if	HW_FLOW_CONTROL_AVAILABLE
 		if (uart0_cb.tx_timeout != SYS_FOREVER_MS) {
-			k_delayed_work_submit(&uart0_cb.tx_timeout_work,
-					      K_MSEC(uart0_cb.tx_timeout));
+			k_timer_start(&uart0_cb.tx_timeout_timer,
+				      K_MSEC(uart0_cb.tx_timeout),
+				      K_MSEC(uart0_cb.tx_timeout));
```

Comment:
shouldn't here also be `K_NO_WAIT`?

Reply:
also this timeout i handled differently than the one in uarte. Here timeout is reset on each byte received and in uarte it is set once before starting.

Reply:
> shouldn't here also be `K_NO_WAIT`?

Of course it should. And it is so in `uart_nrfx_uarte.c`. How did I manage to copy this code incorrectly? Shame on me.

Reply:
Fixed.

Reply:
> also this timeout i handled differently than the one in uarte. Here timeout is reset on each byte received and in uarte it is set once before starting.

Yes. The timer should be rather started in `uart_nrfx_tx()`, not in `tx_isr()` before sending consecutive bytes. The description of API function `uart_tx()` does not state it explicitly, but it seems reasonable to use the timeout value for the whole transaction, not between individual bytes. And [this](https://github.com/zephyrproject-rtos/zephyr/blob/e7df25c848c6ab54bc81430605454f91ca204623/include/drivers/uart.h#L92-L93) note seems to be confirming that.
But I'm not sure if I should try to correct it in this PR.

Reply:
imo, should be fixed separately as this PR focuses on replacing k_delayed_work